web/packages/teleport/src/Notifications/Notifications.tsx

```diff
@@ -0,0 +1,365 @@
+    useState('');
+
+  const [notifications, setNotifications] = useState<NotificationType[]>([]);
+
+  const [userLastSeenNotification, setUserLastSeenNotification] =
+    useState<Date>();
+
+  function fetchNotifications(): Promise<any> {
+    return run(() =>
+      ctx.notificationService
+        .fetchNotifications({
+          clusterId,
+          userNotificationsStartKey: userNotificationsStartKey || undefined,
+          globalNotificationsStartKey: globalNotificationsStartKey || undefined,
+          limit: PAGE_SIZE,
+        })
+        .then(resp => {
+          setUserNotificationsStartKey(resp.userNotificationsNextKey);
+          setGlobalNotificationsStartKey(resp.globalNotificationsNextKey);
+          // If we've reached the end of either list, hide the infinite scroll trigger.
+          if (
+            resp.globalNotificationsNextKey === '' &&
+            resp.userNotificationsNextKey === ''
+          ) {
+            setRenderTrigger(false);
+          }
+          setNotifications([...notifications, ...resp.notifications]);
+          setUserLastSeenNotification(resp.userLastSeenNotification);
+        })
+    );
+  }
+
+  useEffect(() => {
+    fetchNotifications();
+  }, []);
+
+  const { setTrigger } = useInfiniteScroll({
+    fetch: fetchNotifications,
+  });
+
+  const [unseenCount, setUnseenCount] = useState(0);
+  const [view, setView] = useState<View>('All');
+  const [open, setOpen] = useState(false);
+  const [renderTrigger, setRenderTrigger] = useState(true);
+
+  const ref = useRefClickOutside<HTMLDivElement>({ open, setOpen });
+
+  useEffect(() => {
+    // Set number of unseen notifications.
+    const newNotifsCount = notifications.filter(notif =>
+      isBefore(userLastSeenNotification, notif.createdDate)
+    );
+    setUnseenCount(newNotifsCount.length);
+  }, [notifications]);
+
+  const filtered = notifications.filter(notif => {
+    if (view === 'All') {
+      return true;
+    }
+    return !notif.clicked;
+  });
+
+  const items = filtered.map(notif => (
+    <Notification notification={notif} key={notif.id} />
+  ));
+
+  return (
+    <NotificationButtonContainer ref={ref} data-testid="tb-note">
+      <HoverTooltip
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'center' }}
+        tipContent="Notifications"
+        css={`
+          height: 100%;
+        `}
+      >
+        <ButtonIconContainer
+          onClick={() => setOpen(!open)}
+          data-testid="tb-note-button"
+          open={open}
+        >
+          {unseenCount > 0 && (
+            <UnseenBadge>{unseenCount >= 9 ? '9+' : unseenCount}</UnseenBadge>
+          )}
+          <NotificationIcon
+            color={open ? 'text.main' : 'text.muted'}
+            size={iconSize}
+          />
+        </ButtonIconContainer>
+      </HoverTooltip>
+
+      <NotificationsDropdown open={open} data-testid="tb-note-dropdown">
```

Comment:
I considered this when I was running into the bug I mentioned in [this comment](https://github.com/gravitational/teleport/pull/40263#discussion_r1562516716) but there were a couple problems with this, firstly t still caused the bug I mentioned above. Secondly, we would need to manually add a lot more styles to get it look and behave the way we want it to as a topbar dropdown, and it would also no longer be in sync with the style and behaviour of the user menu nav topbar dropdown next to it (which uses `Dropdown.tsx`), we'd like our topbar dropdowns to be consistent with each other and having a separate component for them makes this easier.